modules/standard/IO.chpl

```diff
@@ -3300,27 +3292,37 @@ inline proc channel.readwrite(ref x) where !this.writing {
     }
   }
 
+  pragma "no doc"
+  proc channel.writeBytes(x, len:ssize_t, out error:syserr):bool {
+    compilerWarning("This version of writeBytes() is deprecated; " +
+                    "please switch to a throwing version");
+    try {
+      this.writeBytes(x, len);
+    } catch e: SystemError {
+      error = e.err;
+    } catch {
+      error = EINVAL;
+    }
+  }
+
   /*
-     Write a sequence of bytes.
+     Write `numBytes` bytes to this channel from the memory location
+     associated with `val`..
    */
-  proc channel.writeBytes(x, len:ssize_t, out error:syserr):bool {
+  proc channel.writeBytes(ref val, numBytes: integral): bool throws {
+    var err: syserr = ENOERR;
+    var retval = true;
     on this.home {
       try! this.lock();
-      error = qio_channel_write_amt(false, _channel_internal, x, len);
+      err = qio_channel_write_amt(false, _channel_internal, c_ptrTo(val),
+                                  numBytes.safeCast(ssize_t));
       this.unlock();
+      if err {
+        try this._ch_ioerror(err, "in channel.writeBytes");
+        retval = false;
```

Comment:
`retval = false` will automatically be inserted by the compiler, so I don't think this line is needed.

Reply:
I don't see what would make the compiler insert `retval = false`... can you explain?

Reply:
Ah, I misspoke. When an error is thrown, the `lowerErrorHandling` pass sets the return value to a default value in the function epilogue. In this case it'll set the compiler-inserted return value to `false` and ignore `retval` entirely.

As such, it isn't necessary to set `retval = false`.

Reply:
it's basically dead code, like this
``` chapel
proc f() throws {
  var x = 12;
  throw new MyError();
  x = 22; // this is dead code
  return x; // this is also dead code
```

The call site of a call to `f()` doesn't get a chance to examine the return value, since the function didn't return normally (it threw instead).

Reply:
OK, that suggests to me that this function shouldn't really have a return value anymore (and shouldn't ever have) -- does that look right?

Reply:
Oh, yeah I don't think this needs to return a `bool`. The `read` functions do because `false` indicates `EOF`.